managed/services/agents/credentials_source_agent_invoker.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package agents
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/percona/pmm/api/inventorypb"
+	"github.com/percona/pmm/managed/models"
+	"github.com/percona/pmm/managed/utils/logger"
+)
+
+// CredentialsSourceAgentInvoker requests from agent to parse credentials-source passed in request.
+type CredentialsSourceAgentInvoker struct {
```

Comment:
IMHO, it's already in `agents` directory, so `AgentInvoker` suffix is unnecessary. Maybe `CredentialsSourceLoader` or checker?

Reply:
Changed to `CredentialsSourceLoader`